src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+
+        if (Bulldozer.class.equals(type)) {
+            machines = new BulldozerProducer().get();
+        } else if (Truck.class.equals(type)) {
+            machines = new TruckProducer().get();
+        } else if (Excavator.class.equals(type)) {
+            machines = new ExcavatorProducer().get();
+        }
+
+        return new ArrayList<>(machines);
```

Comment:
producers should be `private final` class-level fields
```suggestion
        if (Bulldozer.class.equals(type)) {
            return bulldozerProducer().get();
        } else if (Truck.class.equals(type)) {
            return truckProducer().get();
        } else if (Excavator.class.equals(type)) {
            return excavatorProducer().get();
        }
```